build-system/tasks/todos.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var BBPromise = require('bluebird');
+var gulp = require('gulp-help')(require('gulp'));
+var path = require('path');
+var srcGlobs = require('../config').presubmitGlobs;
+var util = require('gulp-util');
+var through2 = require('through2');
+var request = BBPromise.promisify(require('request'));
+
+var GITHUB_ACCESS_TOKEN = process.env.GITHUB_ACCESS_TOKEN;
+var issueCache = {};
+
+
+/**
+ * Test if a file's contents contains closed TODOs.
+ *
+ * @param {!File} file file is a vinyl file object
+ * @return {Promise<number>} Number of found closed TODOs.
+ */
+function findClosedTodosInFile(file) {
+  var contents = file.contents.toString();
+  var todos = contents.match(/TODO\([^\)]*\)/g);
+  if (!todos || todos.length == 0) {
+    return Promise.resolve(0);
+  }
+
+  var promises = [];
+  for (var i = 0; i < todos.length; i++) {
+    var todo = todos[i];
+    var parts = /TODO\([^\)]*\#(\d*)\)/.exec(todo);
+    var issueId = parts ? parts[1] : null;
+    if (!issueId) {
+      continue;
+    }
+    promises.push(reportClosedIssue(file, issueId, todo));
+  }
+
+  if (promises.length == 0) {
+    return Promise.resolve(0);
+  }
+  return Promise.all(promises).then(results => {
+    return results.reduce(function(acc, v) {
+      return acc + v;
+    }, 0);
+  }).catch(function(error) {
+    util.log(util.colors.red('Failed in', file.path, error, error.stack));
+    return 0;
+  });
+}
+
+
+/**
+ * @param {!File} file file is a vinyl file object
+ * @param {string} issueId
+ * @param {string} todo
+ * @return {!Promise<number>}
+ */
+function reportClosedIssue(file, issueId, todo) {
+  if (issueCache[issueId] !== undefined) {
+    return Promise.resolve(issueCache[issueId]);
+  }
+  return githubRequest('/issues/' + issueId).then(response => {
```

Comment:
This'll make multiple requests to Github. Instead, you could cache this promise (instead of its resoled value).